docs/operators/s3outputmodule.md

```diff
@@ -0,0 +1,199 @@
+          <value>com.datatorrent.common.partitioner.StatelessPartitioner:{N}</value>
+        </property>     
+        ```
+
+        where {ModuleName} is the name of the S3OutputModule and
+              {N} is the number of static partitions.
+        Above lines will partition S3InitateFileUploadOperator statically {N} times. 
+
+-   ***S3BlockUploadOperator***
+    -   Locality of S3BlockUploadOperator with upstream operator must set to PARTITION_PARALLEL for performance benefits by avoiding serialization/deserialization of objects. So, Partitioning of this operator depends on upstream operator which is of type FSInputModule/BlockReader. 
+    
+-   ***S3FileMergerOperator***
+    -   By setting the parameter "mergerCount", S3FileMergerOperator be statically partitioned. This can be achieved by two ways:
+
+    -   (a) Following code can be added to populateDAG(DAG dag, Configuration conf) method of application to statically partitioning S3FileMergerOperator {N} times:
+
+        ```java
+        FSInputModule inputModule = dag.addModule("HDFSInputModule", new FSInputModule());
+        S3OutputTestModule outputModule = dag.addModule("S3OutputModule", new S3OutputTestModule());
+        outputModule.setMergerCount({N});
+        dag.addStream("FileMetaData", inputModule.filesMetadataOutput, outputModule.filesMetadataInput);
+        dag.addStream("BlocksMetaData", inputModule.blocksMetadataOutput, outputModule.blocksMetadataInput).setLocality(DAG.Locality.CONTAINER_LOCAL);
+        dag.addStream("BlocksData", inputModule.messages, outputModule.blockData).setLocality(DAG.Locality.CONTAINER_LOCAL);
+        ```
+
+    -   (b) By setting the parameter in properties file as follows
+
+        ```xml
+        <property>
+          <name>dt.operator.{ModuleName}.prop.mergerCount</name>
+          <value>{N}</value>
+        </property>     
+        ```
+
+        where {ModuleName} is the name of the S3OutputModule and
+              {N} is the number of static partitions.
+        Above lines will partition S3FileMergerOperator statically {N} times. 
+
+#### Dynamic Partitioning
+--------------------------
+
+Dynamic partitioning is a feature of Apex platform which changes the partition of the operator based on certain condition.
+Locality of S3BlockUploadOperator with upstream operator must set to PARTITION_PARALLEL for performance benefits by avoiding serialization/deserialization of objects. So, Dynamic Partitioning of this operator depends on upstream operator which is of type FSInputModule/BlockReader.
+
+From the example application, By setting the maxReaders and minReaders value to FSInputModule, S3BlockUploadOperator dynamically partitioned between minReaders and maxReaders. This can be achieved by two ways:
+(a) Following code can be added to populateDAG(DAG dag, Configuration conf) method of application to dynamically partitioned S3BlockUploadOperator between {N1} and {N2} times:
```

Comment:
The above 2 lines need lot more explanation to make sense to non-expert readers. I'll make some suggestions later.